File: 3600-find-the-k-th-character-in-string-game-i/find-the-k-th-character-in-string-game-i.cpp
class Solution {
public:
    char kthCharacter(int k) {
        string ans = "a";
        while (ans.length() < k) {
            string temp = "";
            for (char c : ans) {
                if (c == 'z') {
                    temp.push_back('a');
                } else {
                    temp.push_back(c + 1);
                }
            }
            ans += temp;
        }
        return ans[k - 1];
    }
};

File: 3600-find-the-k-th-character-in-string-game-i/find-the-k-th-character-in-string-game-i.java
class Solution {
    public char kthCharacter(int k) {
        char result = 'a';
        k -= 1;
        while (k > 0) {
            result = (char) (result + (k % 2));
            if (result == 123) {
                result = 'a';
            }
            k /= 2;
        }
        return result;
    }
}